geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java

```diff
@@ -1692,14 +1719,38 @@ public void initializeMessageDispatcher() throws CacheException {
       }
       Conflatable nextEvent;
       while ((nextEvent = queuedEvents.poll()) != null) {
```

Comment:
Can we refactor this draining of events (with and without synchronization) instead of duplicating code (only difference is the "with synchronization" in the log message).